device/cuda/src/clusterization/clusters_sum.cu

```diff
@@ -0,0 +1,78 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "clusters_sum.hpp"
+#include "traccc/cuda/utils/definitions.hpp"
+
+namespace traccc::cuda {
+namespace kernels {
+__global__ void clusters_sum(
+    cell_container_types::const_view cells_view,
+    vecmem::data::jagged_vector_view<unsigned int> sparse_ccl_indices_view,
+    unsigned int& total_clusters,
+    vecmem::data::vector_view<std::size_t> cluster_prefix_sum_view,
+    vecmem::data::vector_view<std::size_t> clusters_per_module_view) {
+
+    // Get the global index
+    std::size_t idx = threadIdx.x + blockIdx.x * blockDim.x;
+    // Initialize the data on the device
+    cell_container_types::const_device cells_device(cells_view);
+
+    // Ignore if idx is out of range
+    if (idx >= cells_device.size())
+        return;
+
+    // Get the cells from the current module
+    const auto& cells = cells_device.at(idx).items;
+
+    // Vectors used for cluster indices found by sparse CCL
+    vecmem::jagged_device_vector<unsigned int> device_sparse_ccl_indices(
+        sparse_ccl_indices_view);
+    auto cluster_indices = device_sparse_ccl_indices.at(idx);
+    // Run the sparse_ccl algorithm
+    unsigned int n_clusters = detail::sparse_ccl(cells, cluster_indices);
+    // Save the number of clusters found in this module at
+    // the last, extra place in the indices vectors
+    cluster_indices.back() = n_clusters;
+
+    auto prefix_sum = vecmem::device_atomic_ref<unsigned int>(total_clusters)
+                          .fetch_add(n_clusters);
```

Comment:
This is a pretty inefficient way of doing prefix sums. We'll need to see (through profiling) whether this becomes a bottleneck in practice. If it does, we'll need to perform the same kind of block-wide reduction that I described earlier, followed by a local prefix sum.